interfaces/acados_template/acados_template/acados_casadi_ocp_solver.py

```diff
@@ -96,18 +100,28 @@ def __init__(self, ocp: AcadosOcp, with_hessian=False):
         lb_xtraj_node[0][constraints.idxbx_0] = constraints.lbx_0
         ub_xtraj_node[0][constraints.idxbx_0] = constraints.ubx_0
         offset = 0
+        index_map['lam_x_in_lam_w'].append(list(offset + constraints.idxbx_0))
+        offset += dims.nbx_0 +dims.nbu
         for i in range(1, N_horizon):
             lb_xtraj_node[i][constraints.idxbx] = constraints.lbx
             ub_xtraj_node[i][constraints.idxbx] = constraints.ubx
+            index_map['lam_x_in_lam_w'].append(list(offset + constraints.idxbx))
+            offset += dims.nx + dims.nbu
```

Comment:
Here, it should probably be `dims.nu` instead of `dims.nbu`.
In order to test this, you could add an additional control variable in the test problem, which has just a quadratic cost, but no bounds and doesn't influence the state at all.

I think it is better to merge the loops where the bounds are set up, and add some `if`s inside to handle the first and last stage, such that the `offset` variable is more consistent / understandable.
This should be helpful especially, when we want to add slacks later on.
But maybe you can just do this merge when you add the slacks.